build.gradle

```diff
@@ -308,6 +308,7 @@ project('test:integration') {
         compile project(':test:testcommon')
         compile group: 'junit', name:'junit', version: junitVersion
         compile group: 'org.apache.curator', name: 'curator-test', version: apacheCuratorVersion
+        compile group: 'com.github.vladimir-bukhtoyarov', name: 'bucket4j-core', version: bucket4jVersion
```

Comment:
I am not comfortable with adding yet another dependency, simply for solving this problem. We literally have tons of dependencies already.

Reply:
I agree that it is yet another dependency and possibly we could write something on our own. The reason for picking this library is because it is implementing the well-known `token-bucket` rate limiting algorithm and the integration is very simple (just couple of lines of code).